include/ck_tile/ops/grouped_convolution/utils/transform_conv_fwd_to_gemm.hpp

```diff
@@ -1510,6 +1436,282 @@ struct TransformConvFwdToGemm
+                left_args.output_spatial_lengths_[split_dim] = piece_split_info.out_left;
+                left_args.input_left_pads_[split_dim]        = piece_split_info.left_pad_left;
+                left_args.input_right_pads_[split_dim]       = piece_split_info.right_pad_left;
+
+                // LEFT inherits parent's cumulative offset (no change)
+                auto left_input_offset  = current.input_offset;
+                auto left_output_offset = current.output_offset;
+
+                // Create RIGHT piece (parent offset + local offset)
+                auto right_args                               = current.args;
+                right_args.input_spatial_lengths_[split_dim]  = piece_split_info.in_right;
+                right_args.output_spatial_lengths_[split_dim] = piece_split_info.out_right;
+                right_args.input_left_pads_[split_dim]        = piece_split_info.left_pad_right;
+                right_args.input_right_pads_[split_dim]       = piece_split_info.right_pad_right;
+
+                // CRITICAL: RIGHT accumulates offset (parent + local)
+                auto right_input_offset  = current.input_offset + piece_split_info.input_offset;
+                auto right_output_offset = current.output_offset + piece_split_info.output_offset;
+
+                // Push LEFT and RIGHT back to queue with incremented depth
+                split_queue.emplace(
+                    left_args, left_input_offset, left_output_offset, current.depth + 1);
+                split_queue.emplace(
+                    right_args, right_input_offset, right_output_offset, current.depth + 1);
+            }
+        }
+
+        // Launch all pieces from ready_list
+        float ave_time = 0.0f;
+        for(size_t i = 0; i < ready_list.size(); i++)
+        {
+            const auto& piece = ready_list[i];
+
+            // Create kargs for this piece
+            auto piece_kargs = Kernel::MakeKernelArgs(piece.args);
+
+            // Copy Split-N metadata from original kargs
+            piece_kargs.n_splits   = original_kargs.n_splits;
+            piece_kargs.original_n = original_kargs.original_n;
+
+            // Use batch_stride from ORIGINAL kargs (not split piece's)
+            piece_kargs.input_batch_stride  = original_kargs.input_batch_stride;
+            piece_kargs.output_batch_stride = original_kargs.output_batch_stride;
+
+            // Store cumulative spatial offset (applied per-batch in kernel)
+            piece_kargs.spatial_offset_in  = piece.input_offset;
+            piece_kargs.spatial_offset_out = piece.output_offset;
+
+            const dim3 grids  = Kernel::GridSize(piece_kargs);
+            const dim3 blocks = Kernel::BlockSize();
+
+            if(!Kernel::IsSupportedArgument(piece_kargs))
+            {
+                throw std::runtime_error("Wrong! Split piece arguments not supported!\n");
+            }
+
+            float piece_time = ck_tile::launch_kernel(
```

Comment:
We need to change it to grouped gemm approach since there is possiblity that not all CU are utilized and we can run a few pieces in the parallel